### ISSUE TITLE
Regression with Gradle 8.2-rc-2

### ISSUE THREAD

**Issue by marcphilipp** [bndtools/bnd#5695]
After updating the JUnit build from Gradle 8.1 to 8.2-rc-2 (see [draft PR](https://github.com/junit-team/junit5/pull/3351)), the produced manifests no longer include version numbers for imported packages which causes our `verifyOSGi` `Resolve` tasks to fail ([Build Scan](https://ge.junit.org/s/4etbacry2q3fo/failure#1)).

Full diff:
```diff
--- MANIFEST-8.1	2023-06-12 09:41:32
+++ MANIFEST-8.2	2023-06-12 09:41:32
@@ -1,7 +1,7 @@
 Manifest-Version: 1.0
 Build-Date: 2023-06-12
-Build-Revision: e41a174f53d543ecfee683cf854e61067d7bf900 
-Build-Time: 09:39:13.752+0200
+Build-Revision: 34d235eadd8ab0f318541bce13b6b2ab71d40a28 
+Build-Time: 09:33:40.345+0200
 Built-By: JUnit Team
 Bundle-ManifestVersion: 2
 Bundle-Name: JUnit Platform Suite Commons
@@ -15,12 +15,10 @@
 Implementation-Vendor: junit.org
 Implementation-Version: 1.10.0-SNAPSHOT
 Import-Package: org.apiguardian.api;resolution:=optional;version="[1.1
- ,2)",org.junit.platform.commons.util;version="[1.10,2)",org.junit.pla
- tform.engine;version="[1.10,2)",org.junit.platform.engine.discovery;v
- ersion="[1.10,2)",org.junit.platform.launcher;version="[1.10,2)",org.
- junit.platform.launcher.core;version="[1.10,2)",org.junit.platform.su
- ite.api;version="[1.10,2)",org.junit.platform.commons.logging;status=
- INTERNAL;version="[1.10,2)"
+ ,2)",org.junit.platform.commons.util,org.junit.platform.engine,org.ju
+ nit.platform.engine.discovery,org.junit.platform.launcher,org.junit.p
+ latform.launcher.core,org.junit.platform.suite.api,org.junit.platform
+ .commons.logging;status=INTERNAL
 Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
 Specification-Title: junit-platform-suite-commons
 Specification-Vendor: junit.org
```

Do you have any ideas why that would be?

**Comment by marcphilipp** [comment]
@tresat has written an analysis of the issue [here](https://github.com/junit-team/junit5/pull/3351#issuecomment-1591316685) and will submit a PR

**Comment by pkriens** [comment]
Thanks a lot for doing this. I would really appreciate the PR. 

@bjhargrave time to add your thoughts?

**Comment by bjhargrave** [comment]
Bnd does need to use the jar library elements since Bnd needs the built jars in the class path to (1) access the OSGi metadata in the jar, and (2) the complete contents of the jar since Bnd can assemble a jar quite differently than the contents of the compilation output directories.

**Comment by tresat** [comment]
Yes.  We think the fix to BND will actually be pretty straightforward.  I started work on an integration test in the BND project to demo this situation but haven't finished yet.  Back to that tomorrow.

**Comment by bjhargrave** [comment]
I tested the fix in #5702 against the junit5 build (after reverting [287607c](https://github.com/junit-team/junit5/commit/287607c784930e0baf2bc4bfc61deb6520167330)) and confirmed it fixes the issue.

**Comment by marcphilipp** [comment]
Awesome, thank you!
